### DISCUSSION TITLE
Ground Overlay

### DISCUSSION THREAD

**Discussion by cypher447** [peterLaurence/MapCompose#60] [Q&A] [answered]
Is there a way to place bitmaps on the map and define, where on the map each corner of the bitmap should be?
In OSM there are GroundOverlays to archieve this. Is something similar possible with MapCompose?

**Comment by p-lr** [comment, marked as answer]
You can implement this as a custom composable which scales with the map.
There's an example in the demo app in the "Map with custom drawings" demo. Look at the blue square -- it scales with the map.
The relevant code is in [CustomDraw.kt](https://github.com/p-lr/MapCompose/blob/master/demo/src/main/java/ovh/plrapps/mapcompose/demo/ui/screens/CustomDraw.kt).
You'll see that the `Square` composable uses `DefaultCanvas`, which is done exactly for the purpose of custom composables which need to scale and rotate with the map.

**Reply by cypher447** [reply]
Thanks for the fast answer. It works fine with the code from DefaultCanvas.
I need onClickListeners for the custom draws like they exist for the markers. Is there an approach to do this?
In the demo is shown how to place a marker at a predefined position. Is it also possible to get the position of the map, when a user longClicks (or short) on an empty space on it? I would like to place a marker at the click position.

**Reply by p-lr** [reply]
You can register a click listener on your custom composables using `Modifier.clickable { .. }`
As for having a click listener on the map itself, I believe some new APIs could be added for that purpose. You can make a feature request for that.
